File: 56. Merge Intervals/Solution.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> result;
        if(intervals.size()==1){
            return {intervals[0]};
        }
        sort(intervals.begin(),intervals.end());
        int f1=intervals[0][0], l1=intervals[0][1];
        for(int i=1; i<intervals.size();i++){
            int f=intervals[i][0], l=intervals[i][1];
            if(f<=l1){
                l1 = max(l,l1);
            }else{
                result.push_back({f1,l1});
                f1 = f; 
                l1 = l;
            }
        }
        result.push_back({f1,l1});
        return result;
    }
};

File: 56. Merge Intervals/Solution.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        result = []
        if not intervals:
            return result
        intervals.sort()
        f1,l1 = intervals[0][0],intervals[0][1]
        for i in range(1,len(intervals)):
            f,l = intervals[i][0],intervals[i][1]
            if f<=l1:
                l1 = max(l,l1)
            else:
                result.append([f1,l1])
                f1,l1 = f,l
        result.append([f1,l1])
        return result
